models/userModel.js

```diff
@@ -0,0 +1,236 @@
+import pool from "../db/db.js";
+import bcrypt from "bcrypt";
+import { BadRequestError } from "../expressError.js";
+import {
+  BCRYPT_WORK_FACTOR,
+  REGISTRATION_TTL,
+  JWT_OPTIONS,
+} from "../config.js";
+import crypto from "crypto";
+import base32Encode from "base32-encode";
+import jwt from "jsonwebtoken";
+import { SECRET_KEY } from "../config.js";
+import sendEmailRegistration from "../utils/email.js";
+
+/** Related functions for users */
+
+export default class User {
+  /** Returns userId for email. **/
+
+  static async getId(email) {
+    const userId = await pool.query(
+      `SELECT id
+        FROM users
+        WHERE email = $1`,
+      [email]
+    );
+
+    if (!userId.rows[0]) {
+      throw new BadRequestError(`Invalid email.`);
+    }
+
+    return userId.rows[0].id;
+  }
+
+  /** Register a new account for user.
+   *
+   * Accepts { firstName, lastName, phone, email, password}.
+   * Returns { id, firstName, lastName, phone, email, plainTextToken }.
+   *
+   * Throws BadRequestError on duplicate user emails. **/
+  static async registerAccount({
+    firstName,
+    lastName,
+    phone,
+    email,
+    password,
+  }) {
+    //TODO delete old code below once I get Lawrence's feedback.
+    // const duplicateCheck = await pool.query(
+    //   `SELECT email FROM users
+    //   WHERE email = $1`,
+    //   [email]
+    // );
+
+    // if (duplicateCheck.rows[0]) {
+    //   throw new BadRequestError(
+    //     `Your ${email} is already registered or pending registration.`
+    //   );
+    // }
+
+    /**
+     * //PEER Lawrence, a new hashed password is created before
```

Comment:
Do all read ops like checking validation before write ops

Reply:
Normally, would write ops be separate queries from read ops?

Reply:
Also, when you say read ops, are you talking only about database operations or other types of processing?

Reply:
Lawrence, can you please expand on this feedback? Do you mean that I should perform a duplicate check before attempting to insert data into the DB? If so, would you recommend I write 2 separate queries, which could be more expensive?